src/GafferScene/PathFilter.cpp

```diff
@@ -82,6 +86,31 @@ const Gaffer::ObjectPlug *PathFilter::pathMatcherPlug() const
 	return getChild<Gaffer::ObjectPlug>( g_firstPlugIndex + 1 );
 }
 
+void PathFilter::plugDirtied( const Gaffer::Plug *plug )
+{
+	if( plug == pathsPlug() )
+	{
+		//\todo: share this logic with Switch::variesWithContext()
+		Plug* sourcePlug = pathsPlug()->source<Plug>();
+		if( sourcePlug->direction() == Plug::Out && IECore::runTimeCast<const ComputeNode>( sourcePlug->node() ) )
+		{
+			// pathsPlug() is receiving data from a plug whose value is context varying, meaning 
+			// we need to use the intermediate pathMatcherPlug() in computeMatch() instead:
+			
+			m_pathMatcher = 0;
```

Comment:
Let's use NULL here. I'm slowly getting rid of 0 throughout the codebase so we can replace NULL->nullptr in the future.